services/cega-mock/main_test.go

```diff
@@ -0,0 +1,99 @@
+package main
+
+import (
+	"encoding/base64"
+	"encoding/json"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+)
+
+func TestLoadUsers(t *testing.T) {
+	loadUsers()
+	if len(users) == 0 {
+		t.Fatal("Expected users to be loaded")
+	}
+	if len(usernames) == 0 || len(uids) == 0 {
+		t.Fatal("Expected username and uid maps to be populated")
+	}
+}
+
+func newAuthorizedRequest(method, path, username, password string) *http.Request {
+	req, _ := http.NewRequest(method, path, nil)
+	auth := base64.StdEncoding.EncodeToString([]byte(username + ":" + password))
+	req.Header.Set("Authorization", "Basic "+auth)
+	return req
+}
+
+func TestUserHandler_ValidUsername(t *testing.T) {
+	loadUsers()
+	instances = map[string]string{
+		"test_instance": "test_password",
+	}
+
+	req := newAuthorizedRequest("GET", "/lega/v1/legas/users/dummy?idType=username", "test_instance", "test_password")
```

Comment:
Looks like oldstyle, from the new config of the Proxy service (in test), the auth url is:

CEGA_AUTH_URL=https://nss.test.ega-archive.org/username/

Reply:
I'll replace current routes to the /username one then.